gensim/models/ldacorrelatedmodel.py

```diff
@@ -225,10 +254,17 @@ def do_mstep(self, rho, other, extra_pass=False):
         logger.debug("updating topics")
         # update self with the new blend; also keep track of how much did
         # the topics change through this update, to assess convergence
-        previous_Elogbeta = self.state.get_Elogbeta()
+        previous_Elogbeta = self.get_Elogbeta(self.lambda_kv, self.nu2_kv)
         self.state.blend(rho, other)
 
-        current_Elogbeta = self.state.get_Elogbeta()
+        for k in range(self.num_topics):
+            self.optimize_zeta_for_topic(k)
+            self.optimize_lambda_for_topic(k)
+            self.optimize_zeta_for_topic(k)
+            self.optimize_nu2_for_topic(k)
+            self.optimize_zeta_for_topic(k)
```

Comment:
what's the impact of doing the zeta optimization three times here? we could run a simple check on the gradient differences, I suspect it will be enough to run the optimization once in the beginning and then update all other vars

Reply:
I mainly based this decision on [code](https://github.com/lewer/gensim/blob/0a51e0a37a22e671112f1b371ef891dc04b68b1d/gensim/models/ctmmodel.py#L272) of Correlated Topic Model from unmerged PR to Gensim.
I think you are right - it is not necessary and also increase optimisation time.
I made commit, all tests passed.